Intact.BuinessLogic/Services/ProfilesService.cs

```diff
@@ -60,4 +70,22 @@ public async Task DeleteAsync(Guid userId, string name, CancellationToken cancel
         profileDao.State = ProfileState.Deleted;
         await _appDbContext.SaveChangesAsync(cancellationToken);
     }
+
+    public async Task PickAsync(Guid userId, string name, CancellationToken cancellationToken)
+    {
+        var profileDao = await _appDbContext.Profiles.FindAsync(new object?[] { userId, name }, cancellationToken: cancellationToken);
+        if (profileDao is null)
+            throw new KeyNotFoundException();
+        
+        profileDao.LastPlayed = DateTime.UtcNow;
+        await _redisCache.AddAsync(cacheSet, userId.ToString(), profileDao);
+        await _appDbContext.SaveChangesAsync(cancellationToken);
+    }
+
+    public Task<Profile?> GetCurrentAsync(string userId)
+    {
+        return _redisCache.GetAsync<Profile>(cacheSet, userId);
```

Comment:
maybe a fallback to DB should be used if not found, so we are safe in case of redis down